opal/threads/wait_sync.h

```diff
@@ -75,6 +89,9 @@ static inline int sync_wait_st (ompi_wait_sync_t *sync)
  */
 static inline void wait_sync_update(ompi_wait_sync_t *sync, int updates, int status)
 {
+    sync->signaling = true;
+    /* ensure the signalling value is commited before updating the count */
+    opal_atomic_wmb ();
     if( OPAL_LIKELY(OPAL_SUCCESS == status) ) {
         if( 0 != (OPAL_THREAD_ADD32(&sync->count, -updates)) ) {
```

Comment:
@hjelmn @jladd-mlnx I have a question here.
Consider we have 2 concurrently running wait_sync_updates. Opal thread add is atomic in the sense that intermediate results of the op will not intermix.
**however** we can get both of them receiving 0 if they called this primitive together.
This means another possible race condition:
* Both threads will go to release path
* First thread will release signaling flag
* waiting thread will release sync
* second thread running wait_sync_updates will hit the same bug.